File: backend/src/main/java/taller/multimedia/backend/model/Announcement.java
package taller.multimedia.backend.model;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "announcements")
@Data // Lombok annotation to generate getters, setters, and other utility methods
@NoArgsConstructor
@AllArgsConstructor
public class Announcement {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AnnouncementType type;

    @Column(name = "event_date")
    private LocalDateTime eventDate;

    @Column(name = "location", length = 255)
    private String location;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public Announcement(String title, String content, AnnouncementType type) {
        this.title = title;
        this.content = content;
        this.type = type;
    }
    
}

File: frontend/src/types/announcement.ts
export type AnnouncementType = 'NEWS' | 'EVENT' | 'NOTICE' | 'GENERAL'

// export interface Announcement {
//     id: string
//     title: string
//     content: string
//     type: AnnouncementType
//     eventDate: string | null
//     location: string | null
// }

export interface AnnouncementRequest {
    title: string
    content: string
    type: AnnouncementType
    eventDate?: string
    location?: string
}

export interface Announcement {
    id: string
    title: string
    content: string
    type: AnnouncementType
    eventDate?: string | null
    location?: string | null
}

export interface AnnouncementImageResponse {
    id: string
    announcementId: string
    fileUrl: string
    isCover: boolean
}
